### ISSUE TITLE
Demo fails with failed list all vm azure.BearerAuthorizer#WithAuthorization: Failed to refresh the Token for request

### ISSUE THREAD

**Issue by bq1756** [Azure/aad-pod-identity#618]
**Describe the bug**
I followed the Demo instructions without modification.  Every command deployed what I expected or returned data I expected.  However, the demo pod log shows:

time="2020-05-15T16:28:18Z" level=info msg="starting demo pod demo/demo 10.207.40.24"
time="2020-05-15T16:29:38Z" level=error msg="failed list all vm azure.BearerAuthorizer#WithAuthorization: Failed to refresh the Token for request to https://management.azure.com/subscriptions/2c1b98xxxxxxxx5f1b93f8/resourceGroups/xxRG_NAMExx/providers/Microsoft.Compute/virtualMachines?api-version=2017-12-01: StatusCode=403 -- Original Error: adal: Refresh request failed. Status Code = '403'. Response body: getting assigned identities for pod demo/demo in CREATED state failed after 16 attempts, retry duration [5]s. Error: <nil>\n" podip=10.207.40.24 podname=demo podnamespace=demo
time="2020-05-15T16:30:58Z" level=error msg="failed to refresh ServicePrincipalTokenFromMSI using the MSI VM extension, msiEndpoint(http://169.254.169.254/metadata/identity/oauth2/token)" podip=10.207.40.24 podname=demo podnamespace=demo

**Steps To Reproduce**
export SUBSCRIPTION_ID="2c1b98xxxxxxxx1b93f8"
export RESOURCE_GROUP="xxRG_NAMExx"
export IDENTITY_NAME="demo"
export CLUSTER_RG="MC_xxRG_NAMExx_centralus"
export VMSS="aks-agentpool1-35674147-vmss"
export AKS_CLUSTER_NAME="sa-aks2-nonprod"
export PRINCIPALID="360073xxxxxxxx73c85d"

az role assignment create --role "Managed Identity Operator" --assignee $PRINCIPALID --scope /subscriptions/$SUBSCRIPTION_ID/resourcegroups/$CLUSTER_RG
  - Returned successful JSON

az role assignment create --role "Virtual Machine Contributor" --assignee $PRINCIPALID --scope /subscriptions/$SUBSCRIPTION_ID/resourcegroups/$CLUSTER_RG
  - Returned successful JSON

kubectl apply -f https://raw.githubusercontent.com/Azure/aad-pod-identity/master/deploy/infra/deployment-rbac.yaml
  - created many k8s resources

kubectl apply -f https://raw.githubusercontent.com/Azure/aad-pod-identity/master/deploy/infra/mic-exception.yaml
  - created azurepodidentityexception.aadpodidentity.k8s.io/mic-exception

az identity create -g $RESOURCE_GROUP -n $IDENTITY_NAME --subscription $SUBSCRIPTION_ID
export IDENTITY_CLIENT_ID="$(az identity show -g $RESOURCE_GROUP -n $IDENTITY_NAME --subscription $SUBSCRIPTION_ID --query clientId -otsv)"
  - Returned successful JSON

export IDENTITY_RESOURCE_ID="$(az identity show -g $RESOURCE_GROUP -n $IDENTITY_NAME --subscription $SUBSCRIPTION_ID --query id -otsv)"
  - populated IDENTITY_RESOURCE_ID

export IDENTITY_ASSIGNMENT_ID="$(az role assignment create --role Reader --assignee $IDENTITY_CLIENT_ID --scope /subscriptions/$SUBSCRIPTION_ID/resourceGroups/$RESOURCE_GROUP --query id -otsv)"
  - populated IDENTITY_ASSIGNMENT_ID

cat <<EOF | kubectl apply -f -
apiVersion: "aadpodidentity.k8s.io/v1"
kind: AzureIdentity
metadata:
  name: $IDENTITY_NAME
spec:
  type: 0
  resourceID: $IDENTITY_RESOURCE_ID
  clientID: $IDENTITY_CLIENT_ID
EOF

  - azureidentity.aadpodidentity.k8s.io/demo created

cat << EOF | kubectl apply -f -
apiVersion: v1
kind: Pod
metadata:
  name: demo
  labels:
    aadpodidbinding: $IDENTITY_NAME
spec:
  containers:
  - name: demo
    image: mcr.microsoft.com/k8s/aad-pod-identity/demo:1.2
    args:
      - --subscriptionid=$SUBSCRIPTION_ID
      - --clientid=$IDENTITY_CLIENT_ID
      - --resourcegroup=$RESOURCE_GROUP
    env:
      - name: MY_POD_NAME
        valueFrom:
          fieldRef:
            fieldPath: metadata.name
      - name: MY_POD_NAMESPACE
        valueFrom:
          fieldRef:
            fieldPath: metadata.namespace
      - name: MY_POD_IP
        valueFrom:
          fieldRef:
            fieldPath: status.podIP
  nodeSelector:
    kubernetes.io/os: linux
EOF

  - pod/demo created

**Expected behavior**
I expected the logs to read:
`
succesfully doARMOperations vm count 1
succesfully acquired a token using the MSI, msiEndpoint(http://169.254.169.254/metadata/identity/oauth2/token)
succesfully acquired a token, userAssignedID MSI, msiEndpoint(http://169.254.169.254/metadata/identity/oauth2/token) clientID(xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx)
succesfully made GET on instance metadata
`

**Other verifications**
Verified the AKS cluster runs with MSI
"servicePrincipalProfile": {
    "clientId": "msi"
  }

verified via Azure Portal that MSI "demo" was created 

**Troubleshooting I tried**
I noticed the error message says "podip=10.207.40.24 podname=demo *podnamespace=demo*".  I originally deployed the demo pod to the *default* namespace.  As a test, I also created a demo namespace and deployed a demo pod to the demo namespace.  The demo/demo pod shows the same error as above.

**AAD Pod Identity version**
1.6.0

**Kubernetes version**
1.16.7

**Additional context**

**Comment by chewong** [comment]
Based on the logs, it looks like you are creating the user-assigned identity in `$RESOURCE_GROUP` but you are assigning "Managed Identity Operator" and "Virtual Machine Contributor" to the `$CLUSTER_RG` resource group scope. Could you try creating the user-assigned identity in `$CLUSTER_RG` instead of `$RESOURCE_GROUP`?

**Comment by aramase** [comment]
@bq1756 Did you deploy the `AzureIdentityBinding`? - https://github.com/Azure/aad-pod-identity#5-deploy-azureidentitybinding

If yes, can you check the MIC logs to see why it's failing to assign the identity? If the error is `LinkedAuthorizationFailed`, then all the required role assignments are not provided. Please review this doc and make all the required role assignments. - https://github.com/Azure/aad-pod-identity/blob/master/docs/readmes/README.role-assignment.md

**Comment by bq1756** [comment]
I have the demo working!!!!  I made 2 changes to get it working.

**My First Change**
I changed the `userAssignedIdentities` value that I was using.  Specifically, I was incorrectly supplying `principalId`.  The role assignment wants `clientid`.  In my opinion, the documentation misled me.  I recommend updates to the documentation.

Documentation change?:
[Performing Role Assignments](https://github.com/Azure/aad-pod-identity/blob/master/docs/readmes/README.role-assignment.md#performing-role-assignments) shows `--assignee <PrincipalID>`.  This is confusing.  Because `principalId` does not work.  I used the value of `clientId` here to get the demo to work.

**My Second Change**
I had to create the role assignments and the user-assigned identity in the same resource group, i.e. $CLUSTER_RG.  Is this a limitation of aad-pod-identity or a restriction in Azure?  If this is a aad-pod-identity limitation, I recommend building out the ability to utilize multiple resource groups.

Also, this issue too could benefit from more concise documentation.  [Performing Role Assignments](https://github.com/Azure/aad-pod-identity/blob/master/docs/readmes/README.role-assignment.md#performing-role-assignments) shows `<ClusterResourceGroup>`.  [Demo](https://github.com/Azure/aad-pod-identity#demo)  shows `<ResourceGroup>`.  It would be nice if the names were consistent across the documentation.
